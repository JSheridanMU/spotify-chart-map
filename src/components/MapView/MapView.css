.leaflet-container {
    background-color:#181717;
}

.map-container {
    width: 38vw !important;
}