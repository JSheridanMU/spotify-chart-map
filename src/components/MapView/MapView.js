import React, {Component} from 'react';
import {MapContainer, GeoJSON, ImageOverlay} from 'react-leaflet'
import mapData from '../../data/countries.json'
import codes from '../../data/codes.json'
import 'leaflet/dist/leaflet.css'
import './MapView.css'
import Container from 'react-bootstrap/Container'

const mapStyle = {
    height: '70vh',
    width: '35vw',
};

const bounds = mapData.features[16].geometry.coordinates

class MapView extends Component {
    state = {}

    getCountryCode(threeDigit){
        let countryCodes = Object.assign({}, ...Object.entries(codes).map(([a,b]) => ({ [b]: a })))
        return(countryCodes[threeDigit].toLowerCase())
    }

    onEachCountry = (country, layer) => {
        const countryName = country.properties.ADMIN
        layer.on({
            click: (e) => {
                console.log(countryName ,this.getCountryCode(e.target.feature.properties.ISO_A3))
            }
        })
    }

    render() {
        return(
            <Container className="map-container">
                <MapContainer style={mapStyle} center={[55,20]} zoom={4}>
                    <GeoJSON data={mapData.features} onEachFeature={this.onEachCountry}>
                        <ImageOverlay 
                            url="https://i.pinimg.com/originals/b4/75/00/b4750046d94fed05d00dd849aa5f0ab7.jpg"
                            bounds={bounds}
                        />
                    </GeoJSON>
                </MapContainer>
            </Container>
        )
    }
}

export default MapView