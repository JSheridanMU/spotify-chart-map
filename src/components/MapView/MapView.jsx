import React, {Component} from 'react';
import {MapContainer, GeoJSON} from 'react-leaflet'
import mapData from './../../data/countries.json'
import 'leaflet/dist/leaflet.css'
import './MapView.css'
import Container from 'react-bootstrap/Container'

const mapStyle = {
    height: '90vh',
    width: '100%',
};

class MapView extends Component {
    state = {}
      
    componentDidMount() {
        console.log(mapData)
    }

    render() {
        return(
            <Container className="map-container">
                <MapContainer style={mapStyle} center={[55,20]} zoom={4}>
                    <GeoJSON data={mapData.features}/>
                </MapContainer>
            </Container>
        )
    }
}

export default MapView