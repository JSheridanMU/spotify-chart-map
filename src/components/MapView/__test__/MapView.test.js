import * as React from "react";
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'
import MapView from '.././MapView';

afterEach(cleanup)

test('renders without crashing', () => {
  render(<MapView/>);
});

test('detects watermark test', () => {
  const {getByText} = render(<MapView/>);

  getByText('Leaflet')
});

test('matches snapshot', () => {
    const tree = renderer.create(<MapView/>).toJSON()
    expect(tree).toMatchSnapshot()
});