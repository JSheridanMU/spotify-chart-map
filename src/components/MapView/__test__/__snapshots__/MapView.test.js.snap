// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="map-container container"
>
  <div
    style={
      Object {
        "height": "70vh",
        "width": "35vw",
      }
    }
  />
</div>
`;
